crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -138,23 +139,13 @@ impl Settlement {
     }
 }
 
-const MAINNET_BLOCK_TIME: u64 = 13_000; // ms
-const GNOSIS_BLOCK_TIME: u64 = 5_000; // ms
-const SEPOLIA_BLOCK_TIME: u64 = 13_000; // ms
-const ARBITRUM_ONE_BLOCK_TIME: u64 = 100; // ms
-
 /// How old (in terms of blocks) a settlement should be, to be considered as a
 /// settlement from another environment.
 ///
 /// Currently set to ~6h
-fn max_settlement_age(chain: &infra::blockchain::Id) -> u64 {
+fn max_settlement_age(chain: &Chain) -> u64 {
     const TARGET_AGE: u64 = 6 * 60 * 60 * 1000; // 6h in ms
-    match chain {
-        infra::blockchain::Id::Mainnet => TARGET_AGE / MAINNET_BLOCK_TIME,
-        infra::blockchain::Id::Gnosis => TARGET_AGE / GNOSIS_BLOCK_TIME,
-        infra::blockchain::Id::Sepolia => TARGET_AGE / SEPOLIA_BLOCK_TIME,
-        infra::blockchain::Id::ArbitrumOne => TARGET_AGE / ARBITRUM_ONE_BLOCK_TIME,
-    }
+    chain.blocks_in(TARGET_AGE).round() as u64
```

Comment:
nit: `blocks_in` is only used in autopilot. Just a personal preference to put ONLY COMMON functionalities in `chain` crate.

Reply:
Yeah, not a big deal for this function in particular. Just wanted to align what we put in `chain`, as I can forsee that `blocks_in` will be removed in a few months and that would lead to updating `chain` crate we might want to avoid (if we want it to change only when new chain is added).